model/src/main/kotlin/TextLocation.kt

```diff
@@ -90,4 +92,20 @@ data class TextLocation(
             linesOverlapWith(other) -> 0
             else -> min(abs(other.startLine - endLine), abs(startLine - other.endLine))
         }
+
+    /**
+     * Return a [TextLocation] whose path is relative to [basePath], or throw an [IllegalArgumentException] if the paths
+     * have different roots.
+     */
+    fun withRelativePath(basePath: File): TextLocation {
+        val pathAsFile = File(path)
+        if (!pathAsFile.isAbsolute) return this
```

Comment:
Should this branch also call `invariantSeparatorsPath`?

Reply:
Good catch, the corresponding code in `CommandLinePathScannerWrapper` did that as well. Fixed.